src/Http/Routing/test/UnitTests/Builder/RequestDelegateEndpointRouteBuilderExtensionsTest.cs

```diff
@@ -233,9 +227,11 @@ public void MapEndpoint_PrecedenceOfMetadata_BuilderMetadataReturned()
         var dataSource = Assert.Single(builder.DataSources);
         var endpoint = Assert.Single(dataSource.Endpoints);
 
+        // As with the Delegate Map method overloads for route handlers, the attributes on the RequestDelegate
+        // can override the HttpMethodMetadata. Extension methods could already do this.
         Assert.Equal(3, endpoint.Metadata.Count);
-        Assert.Equal("ATTRIBUTE", GetMethod(endpoint.Metadata[0]));
-        Assert.Equal("METHOD", GetMethod(endpoint.Metadata[1]));
+        Assert.Equal("METHOD", GetMethod(endpoint.Metadata[0]));
+        Assert.Equal("ATTRIBUTE", GetMethod(endpoint.Metadata[1]));
```

Comment:
Breaking change?

Reply:
Technically.

You'd had to have defined a custom attribute that implemented `IHttpMethodMetadata` which [no one on grepp.app has](https://grep.app/search?q=IHttpMethodMetadata) other than us in this very test, [in one other test](https://github.com/dotnet/aspnetcore/blob/main/src/Http/Routing/test/testassets/RoutingSandbox/UseEndpointRoutingStartup.cs#L80) which notably didn't need to be changed because it wasn't overriding anything, and in .NET 6 preview diffs before I removed it from the MVC attributes. We certainly don't ship any such attributes.

Plus, we already made this "breaking" change for route handlers between .NET 6 and .NET 7 with #40926. See https://github.com/dotnet/aspnetcore/pull/40926/files#diff-f6727c82b3f6cff19c30c9c793f3e47c1c2ff2035a6453198ff4cedad8e84557

I almost went the other way and had `RouteEndopintDataSource` copy the old .NET 6 behavior here so we didn't "break" anything for either `Delegate` or `RequestDelegate`, but it makes sense to configure metadata added implicitly by the `MapXxx` call like the `MethodInfo` and `HttpMethodMetadata` before any additional metadata configured by the user.